### PR TITLE
feat(pod):  added softDelete, onDelete and onUndo functionalities - FE-4165

### PR DESCRIPTION
**Pull request by mkrds**
### Proposed behaviour
This PR adds softDelete, onDelete and onUndo functionalities to Pod and ShowEditPod

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

**Review by sjohnson-sage — COMMENTED** [review]
LGTM in general, you need to modify the TypeScript files to match the js ones though

File: src/components/pod/pod.style.js [hunk 1]
```diff
@@ -136,89 +158,124 @@ const StyledEditContainer = styled.div`
...
+  ${isFocused &&
+  css`
+    outline: 3px solid ${theme.colors.focus};
+    border: none;
+    padding: ${actionButtonPaddings[size] +
+    (noBorder || internalEditButton ? 0 : 1)}px;
+  `};
+`;
+
+const StyledEditContainer = styled.div``;
```
Comment (sjohnson-sage) [inline src/components/pod/pod.style.js:234]: Might as well get rid of this styled component as it's not adding any value

**Review by sjohnson-sage — DISMISSED** [review]

File: src/components/pod/pod.spec.js [hunk 2]
```diff
@@ -834,6 +1013,14 @@ function renderEditAction(props = {}, renderer = mount) {
   return renderer(<StyledEditAction {...props} />);
 }
 
+function renderDeleteAction(props = {}, renderer = mount) {
+  return renderer(<StyledDeleteButton {...props} />);
```
Comment (edleeks87) [inline src/components/pod/pod.spec.js:1017]: These two new functions throw errors in the console when running the tests when no `onAction` or `children` are passed

**Review by edleeks87 — COMMENTED** [review]
I also noticed that `height` has an incorrect prop definition, would you mind updating it in this PR as it's a one line fix?

`PropTypes.oneOf` -> `PropTypes.oneOfType`

https://github.com/Sage/carbon/blob/351b5991f5e983e97149becdc5453267385130d8/src/components/pod/pod.component.js#L533

**Review by edleeks87 — DISMISSED** [review]

**Review by sjohnson-sage — DISMISSED** [review]

**Review by edleeks87 — APPROVED** [review]

File: src/components/pod/pod.d.ts [hunk 3]
```diff
@@ -30,7 +24,22 @@ export interface PodProps {
...
+    | string
+    | {}
+    | ((
+        ev: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>
+      ) => void);
+  /** Supplies a delete action to the pod */
+  onDelete?: (
+    ev: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>
+  ) => void;
+  /** Supplies an edit action to the pod */
```
Comment (DipperTheDan) [inline src/components/pod/pod.d.ts:37]: Shouldn't this be `Supplies an undo action to the pod` or something similar?

**Review by edleeks87 — APPROVED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]
Approve for @ZhuoyuJin from QA

Cypress tests will be covered in `FE-4249`

**Review by edleeks87 — APPROVED** [review]

**Pull request merged**